client/signup/steps/site-type/form.jsx

```diff
@@ -20,19 +24,23 @@ import './style.scss';
 
 class SiteTypeForm extends Component {
 	static propTypes = {
-		siteType: PropTypes.string,
 		submitForm: PropTypes.func.isRequired,
 
 		// from localize() HoC
 		translate: PropTypes.func.isRequired,
 	};
 
-	constructor( props ) {
-		super( props );
-		this.state = {
-			siteType: props.siteType,
```

Comment:
Hm, the WP.com step still relies on this, so we shouldn't be removing it:

https://github.com/Automattic/wp-calypso/blob/ad2e4bdb62ead2751cc40e5c14a6e30b3195b6d2/client/signup/steps/site-type/index.jsx#L53